fdbclient/NativeAPI.actor.cpp

```diff
@@ -3690,8 +3690,21 @@ ACTOR Future<StorageMetrics> getStorageMetricsLargeKeyRange(Database cx, KeyRang
 	state int nLocs = locations.size();
 	state vector<Future<StorageMetrics>> fx(nLocs);
 	state StorageMetrics total;
+	KeyRef partBegin, partEnd;
 	for (int i = 0; i < nLocs; i++) {
-		fx[i] = doGetStorageMetrics(cx, locations[i].first, locations[i].second);
+		if (i == 0) {
+			// Use the actual begin key instead of the shard begin
+			partBegin = keys.begin;
+		} else {
+			partBegin = locations[i].first.begin;
```

Comment:
Would this be slightly shorter?
partBegin = (i == 0) ? keys.begin : locations[i].first.begin;;
partEnd = (i == nLocs - 1) ? keys.end : locations[i].first.end;

Reply:
Done.